sequencer/src/state.rs

```diff
@@ -1,413 +1,26 @@
-        });
-    }
-
-    let ValidatedState {
-        block_merkle_tree,
-        fee_merkle_tree,
-        ..
-    } = state;
-
-    let block_merkle_tree_root = block_merkle_tree.commitment();
-    if proposal.block_merkle_tree_root != block_merkle_tree_root {
-        return Err(ProposalValidationError::InvalidBlockRoot {
-            expected_root: block_merkle_tree_root,
-            proposal_root: proposal.block_merkle_tree_root,
-        });
-    }
-
-    let fee_merkle_tree_root = fee_merkle_tree.commitment();
-    if proposal.fee_merkle_tree_root != fee_merkle_tree_root {
-        return Err(ProposalValidationError::InvalidFeeRoot {
-            expected_root: fee_merkle_tree_root,
-            proposal_root: proposal.fee_merkle_tree_root,
-        });
-    }
-
-    proposal
-        .ns_table
-        .validate(&PayloadByteLen::from_vid_common(vid_common))?;
-
-    Ok(())
-}
-
-/// Possible charge fee failures
-#[derive(Error, Debug, Eq, PartialEq)]
-pub enum FeeError {
-    #[error("Insuficcient Funds: have {balance:?}, required {amount:?}")]
-    InsufficientFunds {
-        balance: Option<FeeAmount>,
-        amount: FeeAmount,
-    },
-    #[error("Merkle Tree Error: {0}")]
-    MerkleTreeError(MerkleTreeError),
-}
-
-impl From<MerkleTreeError> for FeeError {
-    fn from(item: MerkleTreeError) -> Self {
-        Self::MerkleTreeError(item)
-    }
-}
-
-fn charge_fee(
-    state: &mut ValidatedState,
-    delta: &mut Delta,
-    fee_info: FeeInfo,
-    recipient: FeeAccount,
-) -> Result<(), FeeError> {
-    state.charge_fee(fee_info, recipient)?;
-    delta.fees_delta.extend([fee_info.account, recipient]);
-    Ok(())
-}
-
-/// Possible builder validation failures
-#[derive(Error, Debug, Eq, PartialEq)]
-pub enum BuilderValidationError {
-    #[error("Builder signature not found")]
-    SignatureNotFound,
-    #[error("Fee amount out of range: {0}")]
-    FeeAmountOutOfRange(FeeAmount),
-    #[error("Invalid Builder Signature")]
-    InvalidBuilderSignature,
-}
-
-/// Validate builder account by verifying signature
-fn validate_builder_fee(proposed_header: &Header) -> Result<(), BuilderValidationError> {
-    // Beware of Malice!
-    let signature = proposed_header
-        .builder_signature
-        .ok_or(BuilderValidationError::SignatureNotFound)?;
-    let fee_amount = proposed_header.fee_info.amount().as_u64().ok_or(
-        BuilderValidationError::FeeAmountOutOfRange(proposed_header.fee_info.amount()),
-    )?;
-
-    // verify signature
-    if !proposed_header.fee_info.account.validate_fee_signature(
-        &signature,
-        fee_amount,
-        proposed_header.metadata(),
-        &proposed_header.payload_commitment(),
-    ) {
-        return Err(BuilderValidationError::InvalidBuilderSignature);
-    }
-
-    Ok(())
-}
+use crate::{
+    api::data_source::CatchupDataSource, catchup::SqlStateCatchup,
+    persistence::ChainConfigPersistence, NodeState, SeqTypes,
+};
 
 async fn compute_state_update(
```

Comment:
Are we keeping the `src/*` stuff around for pure utility functions like these. Just trying to understand the relationship between things like `src/state.rs` and `impls/state.rs`.